internal/store/db/mongo/store_test.go

```diff
@@ -12,4 +14,207 @@
  * the License.
  *******************************************************************************/
 
+// This test will only be executed if the tag mongoRunning is added when running
+// the tests with a command like:
+// go test -tags mongoRunning
+
 package mongo
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/db"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/models"
+
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+var TestHost = "localhost"
+var TestPort = 27017
+var TestTimeout = 5000
+var TestDatabaseName = "test"
+var TestBatchSize = 1337
+
+var TestValidNoAuthConfig = db.Configuration{
+	Type:         db.MongoDB,
+	Host:         TestHost,
+	Port:         TestPort,
+	Timeout:      TestTimeout,
+	DatabaseName: TestDatabaseName,
+	BatchSize:    TestBatchSize,
+}
+
+var TestTimeoutConfig = db.Configuration{
+	Type:         db.MongoDB,
+	Host:         TestHost,
+	Port:         TestPort,
+	Timeout:      1,
+	DatabaseName: TestDatabaseName,
+	BatchSize:    TestBatchSize,
+}
+
+var TestID = primitive.NewObjectID().Hex()
+var TestAppServiceKey = uuid.New().String()
+var TestPayload = []byte("brandon was here")
+var TestPipelinePosition = 0
+var TestVersion = "1"
+
+var TestStoredObject = models.NewStoredObject(TestID, TestAppServiceKey, TestPayload, TestPipelinePosition, TestVersion)
+
+func TestNewClient(t *testing.T) {
+	tests := []struct {
+		name          string
+		config        db.Configuration
+		expectedError bool
+	}{
+		{"Success, no auth", TestValidNoAuthConfig, false},
+		{"Timed out", TestTimeoutConfig, true},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := NewClient(test.config)
+
+			if test.expectedError && err == nil {
+				t.Error("Expected an error")
+				return
+			}
+
+			if !test.expectedError && err != nil {
+				t.Errorf("Unexpectedly encountered error: %s", err.Error())
+				return
+			}
+		})
+	}
+}
+
+func TestClient_CRUD(t *testing.T) {
+	tests := []struct {
+		name             string
+		config           db.Configuration
+		expectedError    bool
+		expectedErrorVal error
+	}{
+		{"Success, no auth", TestValidNoAuthConfig, false, nil},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			client, _ := NewClient(test.config)
```

Comment:
How are you mocking the Mongo DB?

Reply:
There are no mocks in this test.

Reply:
Must have a mock since there will not be a DB instance running when the verify checks run the tests. @tsconn23 should be able to show you how edgex-go creates mocks for the DB layer.

Reply:
We do that for the REST and business logic associated with storing to the database, but right now we have no unit tests for the Mongo or Redis drivers themselves. We have one integration test for Mongo that looks a lot like what I turned in with my first pass.

Reply:
And conceptually, if we mock out the Mongo stuff outside of our control, what are we actually testing?